src/core/parser.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <vector>
+#include <initializer_list>
+#include <nlohmann/json.hpp>
+#include "json_writer.hpp"
+#include "parse_worker.hpp"
+#include "parser_internal/docgen_worker.hpp"
+#include "parser_internal/comment_worker.hpp"
+#include "parser_internal/func_worker.hpp"
+#include "parser_internal/ignore_worker.hpp"
+#include "parser_internal/tag_worker.hpp"
+
+namespace docgen {
+namespace core {
+
+class Parser
+{
+	public:
+		using CommentWorker = parser_internal::CommentWorker;
+		using FuncWorker = parser_internal::FuncWorker;
+		using IgnoreWorker = parser_internal::IgnoreWorker;
+		using TagWorker = parser_internal::TagWorker;
+
```

Comment:
would be nice to have `using symbol_t = ...` and use `symbol_t` instead of `Symbol`.